pennylane/decomposition/resources.py

```diff
@@ -37,7 +37,7 @@ class Resources:
     gate_counts: dict[CompressedResourceOp, int] = field(default_factory=dict)
 
     def __post_init__(self):
-        """Verify that the gate counts and the number of gates are consistent."""
+        """Remove zero-count gates and verify that num_gates is correct."""
```

Comment:
Both the change and the original description confuse me :sweat_smile: Isn't it more like
```suggestion
        """Verify that all gate counts are non-zero."""
```